### PR TITLE
NMS-15439: implement new OPA 1.5.0 alarm ticketing interfaces

### PR DESCRIPTION
**Pull request by RangerRick**
This PR implements the Horizon core side of https://github.com/OpenNMS/opennms-integration-api/pull/79

### External References

* Jira (Issue Tracker): https://opennms.atlassian.net/browse/NMS-15439

### CODE REVIEW

**Comment by RangerRick** [comment]
FYI this will not pass until 1.5.0 is merged, obviously

**Review by christianpape — DISMISSED** [review]
LGTM.

File: features/api-layer/dao-common/src/main/java/org/opennms/features/apilayer/dao/AlarmDaoImpl.java [hunk 1]
```diff
@@ -83,4 +92,101 @@ public Optional<Alarm> getAlarmWithHighestSeverity(NodeRef nodeRef) {
...
+            return Optional.of(alarm);
+        });
+    }
+
+    @Override
+    public void setTicketState(final State state, final int... alarmIds) {
+        sessionUtils.withTransaction(() -> {
+            Arrays.stream(alarmIds).boxed().map(alarmDao::get).forEach(alarm -> {
+                alarm.setTTicketState(ModelMappers.fromTicketState(state));
+                alarmDao.saveOrUpdate(alarm);
```
Comment (j-white) [inline features/api-layer/dao-common/src/main/java/org/opennms/features/apilayer/dao/AlarmDaoImpl.java:117]: Should also call `AlarmEntityNotifier#didChangeTicketStateForAlarm` after `saveOrUpdate`.

File: features/api-layer/dao-common/src/main/java/org/opennms/features/apilayer/dao/AlarmDaoImpl.java [hunk 2]
```diff
@@ -83,4 +92,101 @@ public Optional<Alarm> getAlarmWithHighestSeverity(NodeRef nodeRef) {
...
+    public void setSeverity(final Severity severity, final int... alarmIds) {
+        final var onmsSeverity = ModelMappers.fromSeverity(severity);
+        sessionUtils.withTransaction(() -> {
+            final Criteria criteria = new CriteriaBuilder(OnmsAlarm.class)
+                    .and(Restrictions.in("id", Arrays.asList(alarmIds)))
+                    .toCriteria();
+            final List<OnmsAlarm> alarms = alarmDao.findMatching(criteria);
+            alarms.forEach(alarm -> {
+                alarm.setSeverity(onmsSeverity);
+                alarmDao.saveOrUpdate(alarm);
```
Comment (j-white) [inline features/api-layer/dao-common/src/main/java/org/opennms/features/apilayer/dao/AlarmDaoImpl.java:188]: Should also call `AlarmEntityNotifier#didUpdateAlarmSeverity` after `saveOrUpdate`.

**Review by j-white — COMMENTED** [review]
Looks good - just missing some calls to the `AlarmEntityNotifier` that need to be made after updating the entities. These calls already happen withing the `AckDao` though, so they don't need to be explicit there.

**Comment by RangerRick** [comment]
> @j-white: Looks good - just missing some calls to the `AlarmEntityNotifier` that need to be made after updating the entities. These calls already happen withing the `AckDao` though, so they don't need to be explicit there.

Good catch! For some reason I thought that was already happening on that side of the code.  I'll update it.

**Pull request merged**